### PR TITLE
Azure samples with learning transport/persistence

### PR DESCRIPTION
**Pull request by SeanFeldman**
Total of 4 samples that use either learning transport, persistence, or both.

* [azure-service-bus](https://docs.particular.net/samples/azure/azure-service-bus)
   * [x] Not required
 * [azure-service-bus-long-running](https://docs.particular.net/samples/azure/azure-service-bus-long-running)
   * [x] Persistence
 * [azure-service-fabric-routing](https://docs.particular.net/samples/azure/azure-service-fabric-routing)
   * [x] Not required
 * [blob-storage-databus](https://docs.particular.net/samples/azure/blob-storage-databus)
   * [x] Persistence
   * [x] Transport
 * [custom-partitioning-asb](https://docs.particular.net/samples/azure/custom-partitioning-asb)
   * [x] Not required
 * [custom-sanitization-asb](https://docs.particular.net/samples/azure/custom-sanitization-asb)
   * [x] Not required
 * [native-integration-asb](https://docs.particular.net/samples/azure/native-integration-asb)
   * [x] Not required
 * [performance-tuning-asb](https://docs.particular.net/samples/azure/performance-tuning-asb)
   * [x] Not required
 * [polymorphic-events-asb](https://docs.particular.net/samples/azure/polymorphic-events-asb)
   * [x] Not required
 * [shared-host](https://docs.particular.net/samples/azure/shared-host)
   * [x] Not required
 * [storage-persistence](https://docs.particular.net/samples/azure/storage-persistence)
   * [x] Transport
 * [storage-queues](https://docs.particular.net/samples/azure/storage-queues)
   * [x] Persistence

Addresses https://github.com/Particular/NServiceBus.Persistence.AzureStorage/issues/142
Completes https://github.com/Particular/docs.particular.net/pull/2724

### CODE REVIEW

**Comment by SeanFeldman** [comment]
`shared-host` is entirely based on the storage account and trying to yank the transport (ASQ) out of it makes very little sense. @Particular/azure-maintainers please review **again** the samples to make sure I didn't omit anything.

/cc @SimonCropp

File: samples/azure/blob-storage-databus/ABSDataBus_1/Receiver/Program.cs [hunk 1]
```diff
@@ -16,7 +16,8 @@ static void Main()
         endpointConfiguration.UseSerialization<JsonSerializer>();
         var dataBus = endpointConfiguration.UseDataBus<AzureDataBus>();
         dataBus.ConnectionString("UseDevelopmentStorage=true");
-        endpointConfiguration.UsePersistence<InMemoryPersistence>();
+        endpointConfiguration.UseTransport<LearningTransport>();
+        endpointConfiguration.UsePersistence<LearningPersistence>();
```
Comment (yvesgoeleven) [inline samples/azure/blob-storage-databus/ABSDataBus_1/Receiver/Program.cs]: Is UsePersistence mandatory here? this receiver doesn't seem to use persistance, there are no saga's or does the  learning transport require a persister?

File: samples/azure/blob-storage-databus/ABSDataBus_1/Sender/Program.cs [hunk 2]
```diff
@@ -22,7 +22,8 @@ static void Main()
 
         #endregion
 
-        endpointConfiguration.UsePersistence<InMemoryPersistence>();
+        endpointConfiguration.UseTransport<LearningTransport>();
+        endpointConfiguration.UsePersistence<LearningPersistence>();
```
Comment (yvesgoeleven) [inline samples/azure/blob-storage-databus/ABSDataBus_1/Sender/Program.cs]: Is UsePersistence mandatory here? this receiver doesn't seem to use persistance, there are no saga's and the  learning transport does not require a persister. Imo this line can be removed.

**Comment by yvesgoeleven** [comment]
I think all samples are covered

File: samples/azure/blob-storage-databus/ABSDataBus_1/Receiver/Program.cs (hunk 1 above)
Comment (weralabaj) [inline samples/azure/blob-storage-databus/ABSDataBus_1/Receiver/Program.cs]: it is required, unless we disable timeout manager, but then retries won't be available, so I'd rather leave as is.

**Pull request merged**